src/main/java/org/maxgamer/quickshop/util/language/game/MojangGameLanguageImpl.java

```diff
@@ -186,150 +169,140 @@ public String getBlock(@NotNull Material material) {
 
 @Getter
 @Setter
-//TODO Refactor: After two months what i write it and now i can't understand these code...
 class GameLanguageLoadThread extends Thread {
     private JsonObject lang;
 
     private String languageCode;
 
-    private boolean mainThreadWaiting;
+    private boolean isLatest = false; //Does assets is latest?
+
+    private boolean isUpdated = false; //Did we tried update assets?
 
     private QuickShop plugin;
 
     public void run() {
-        boolean failed = false;
         try {
-            File cacheFile = new File(Util.getCacheFolder(), "lang.cache"); // Load cache file
+            File cacheFile = new File(Util.getCacheFolder(), "mojanglang.cache"); // Load cache file
             if (!cacheFile.exists()) {
                 //noinspection ResultOfMethodCallIgnored
                 cacheFile.createNewFile();
             }
             YamlConfiguration yamlConfiguration = new YamlConfiguration();
-            yamlConfiguration.load(new File(Util.getCacheFolder(), "lang.cache"));
-            boolean needUpdateCache = false;
+            yamlConfiguration.load(cacheFile);
             /* The cache data, if it all matches, we doesn't need connect to internet to download files again. */
-            String cachingServerVersion = yamlConfiguration.getString("ver");
-            String cachingLanguageHash = yamlConfiguration.getString("hash");
-            String cachingLanguageName = yamlConfiguration.getString("lang");
+            //String cacheVersion = yamlConfiguration.getString("ver");
+            String cacheSha1 = yamlConfiguration.getString("sha1", "ERROR");
+            String cacheCode = yamlConfiguration.getString("lang");
             /* If language name is default, use computer language */
             if ("default".equals(languageCode)) {
                 Locale locale = Locale.getDefault();
                 languageCode = locale.getLanguage() + "_" + locale.getCountry();
             }
-            if (!languageCode.equals(cachingLanguageName)) {
-                cachingLanguageName = languageCode;
-                needUpdateCache = true;
-            }
-            String languageCode1 = languageCode.toLowerCase();
-            String serverVersion = ReflectFactory.getServerVersion();
-            if (!serverVersion.equals(cachingServerVersion)) {
-                cachingServerVersion = serverVersion;
-                needUpdateCache = true;
+            if (languageCode.equalsIgnoreCase("en_us")) {
+                isLatest = true;
+                return; //Ignore english language
             }
+            if (languageCode.equals(cacheCode) && new File(Util.getCacheFolder(), cacheSha1).exists()) {
```

Comment:
How about `cacheSha1="ERROR"`?

Reply:
The code won't run cause `ERROR` file not exists.
And I have added hash equals in coming commit